website/integrations: add Homarr (#13818)
goauthentik/authentik: website/integrations/services/homarr/index.md

---
title: Integrate with Homarr
sidebar_label: Homarr
support_level: community
---

## What is Homarr

> A sleek, modern dashboard that puts all of your apps and services at your fingertips. Control everything in one convenient location. Seamlessly integrates with the apps you've added, providing you with valuable information.
>
> -- https://homarr.dev/

## Preparation

The following placeholders are used in this guide:

> - `homarr.company` is the FQDN of the Hoarder installation.

Reviewer: ```suggestion
- `homarr.company` is the FQDN of the Homarr installation.
```

:::

## authentik configuration

To support the integration of Homarr with authentik, you need to create an application/provider pair in authentik.

### Create an application and provider in authentik

1. Log in to authentik as an admin, and open the authentik Admin interface.
2. Navigate to **Applications** > **Applications** and click **Create with Provider** to create an application and provider pair. (Alternatively you can first create a provider separately, then create the application and connect it with the provider.)

- **Application**: provide a descriptive name, an optional group for the type of application, the policy engine mode, and optional UI settings.
- **Choose a Provider type**: select **OAuth2/OpenID Connect** as the provider type.
- **Configure the Provider**: provide a name (or accept the auto-provided name), the authorization flow to use for this provider, and the following required configurations.
    - Note the **Client ID**,**Client Secret**, and **slug** values because they will be required later.

>     - Set a `Regex` redirect URI to <kbd>.*</kbd>.

Reviewer: Can you find a strict regex redirect uri? this value is unsafe

Author: @dominic-r updated with hard coded uris

>     - Create two `strict` redirect URI and set to <kbd>https://homarr.company/api/auth/callback/oidc</kbd>. and <kbd> http://localhost:50575/api/auth/callback/oidc</kbd>

Reviewer 2: ```suggestion
    - Create two `strict` redirect URI and set to <kbd>https://homarr.company/api/auth/callback/oidc</kbd> and <kbd> http://localhost:50575/api/auth/callback/oidc</kbd>
```

    - Select any available signing key.
- **Configure Bindings** _(optional)_: you can create a [binding](/docs/add-secure-apps/flows-stages/bindings/) (policy, group, or user) to manage the listing and access to applications on a user's **My applications** page.

3. Click **Submit** to save the new application and provider.

## Homarr configuration

> In Homarr, you'll need to add these environment variables:

Reviewer 3: ```suggestion
Add the following environment variables to your Homarr configuration. Make sure to fill in the Client ID, Client Secret, OIDC Issuer, and OIDC URI from your authentik instance.
```

```sh
AUTH_PROVIDERS="oidc,credentials"
AUTH_OIDC_CLIENT_ID=<Client ID from authentik>
AUTH_OIDC_CLIENT_SECRET=<Client secret from authentik>
AUTH_OIDC_ISSUER=https://authentik.company/application/o/<slug from authentik>
AUTH_OIDC_URI=https://authentik.company/application/o/authorize
AUTH_OIDC_CLIENT_NAME=authentik
OAUTH_ALLOW_DANGEROUS_EMAIL_ACCOUNT_LINKING=true
# Optional: You can add this if you only want to allow auto login via authentik
# AUTH_OIDC_AUTO_LOGIN=true
```

> Finally, restart the Homarr server and test your configuration.

Reviewer 3: ```suggestion
Restart the Homarr service for the changes to take effect.
```
